airbyte-workers/src/test/java/io/airbyte/workers/temporal/scheduling/activities/FeatureFlagFetchActivityTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling.activities;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import io.airbyte.api.client.generated.ConnectionApi;
+import io.airbyte.api.client.generated.SourceApi;
+import io.airbyte.api.client.invoker.generated.ApiException;
+import io.airbyte.api.client.model.generated.ConnectionIdRequestBody;
+import io.airbyte.api.client.model.generated.ConnectionRead;
+import io.airbyte.api.client.model.generated.SourceIdRequestBody;
+import io.airbyte.api.client.model.generated.SourceRead;
+import io.airbyte.featureflag.FeatureFlagClient;
+import io.airbyte.featureflag.FieldSelectionEnabled;
+import io.airbyte.featureflag.TestClient;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class FeatureFlagFetchActivityTest {
+
+  final static UUID CONNECTION_ID = UUID.randomUUID();
+  final static UUID WORKSPACE_ID = UUID.randomUUID();
+
+  FeatureFlagFetchActivity featureFlagFetchActivity;
+  FeatureFlagClient featureFlagClient;
+
+  @BeforeEach
+  void setUp() throws ApiException {
+    final SourceApi sourceApi = mock(SourceApi.class);
+    final ConnectionApi connectionApi = mock(ConnectionApi.class);
+
+    featureFlagClient = new TestClient(Map.of(FieldSelectionEnabled.INSTANCE.getKey(), true));
```

Comment:
Yeah, agree - I orignially tried to mock it and it was not obvious that I had to use this other TestClient class

Reply:
Actually, I think that it might be mockable with mock(Feature::class.java) (from [here](https://medium.com/21buttons-tech/mocking-kotlin-classes-with-mockito-the-fast-way-631824edd5ba)), I'll try that on my PR.

Reply:
It is possible to mock it if we remove the `sealed` keyword from the interface. @colesnodgrass what is your opinion in removing the `sealed` keyword?

Reply:
@benmoriceau @pedroslopez while you cannot create a mock of the `sealed interface`, you can create a mock of the `TestClient`.

```java
private final FeatureFlagClient featureFlagClient = mock(TestClient.class);

void testFailHeartbeat() {
    when(featureFlagClient.enabled(eq(ShouldFailSyncIfHeartbeatFailure.INSTANCE), any())).thenReturn(true);
    when(heartbeatMonitor.isBeating()).thenReturn(Optional.of(false));
    assertThrows(HeartbeatTimeoutChaperone.HeartbeatTimeoutException.class,
        () -> heartbeatTimeoutChaperone.runWithHeartbeatThread(() -> {
          try {
            Thread.sleep(Long.MAX_VALUE);
          } catch (final InterruptedException e) {
            throw new RuntimeException(e);
          }
        }, executorService).run());
  }